File: sdks/typescript/src/types/evaluations.ts
export interface EvalCase {
  query: string;
  relevant_ids: string[];
  top_k?: number | null;
}

export interface EvalBody {
  collection: string;
  cases: EvalCase[];
  top_k?: number;
  search_mode?: "semantic" | "keyword" | "hybrid";
}

export interface EvalPerCase {
  query: string;
  hit_ids: string[];
  expected_ids: string[];
  recall_at_k: number;
  reciprocal_rank: number;
  ndcg_at_k: number;
}

export interface EvalResponse {
  collection: string;
  total_cases: number;
  recall_at_k_avg: number;
  mrr: number;
  ndcg_at_k_avg: number;
  per_case: EvalPerCase[];
}

File: sdks/rust/src/types/evaluations.rs
use serde::{Deserialize, Serialize};

use crate::types::query::SearchMode;

/// Evaluation case.
#[derive(Debug, Clone, Serialize)]
pub struct EvalCase {
    /// Query text.
    pub query: String,
    /// Relevant document or chunk IDs.
    pub relevant_ids: Vec<String>,
    /// Per-case top-k override.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub top_k: Option<u32>,
}

/// Evaluation request body.
#[derive(Debug, Clone, Serialize)]
pub struct EvalBody {
    /// Collection name.
    pub collection: String,
    /// Evaluation cases.
    pub cases: Vec<EvalCase>,
    /// Default top-k.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub top_k: Option<u32>,
    /// Search mode.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub search_mode: Option<SearchMode>,
}

/// Per-case evaluation result.
#[derive(Debug, Clone, Deserialize)]
pub struct EvalPerCase {
    /// Query text.
    pub query: String,
    /// Hit IDs.
    pub hit_ids: Vec<String>,
    /// Expected IDs.
    pub expected_ids: Vec<String>,
    /// Recall at k.
    pub recall_at_k: f64,
    /// Reciprocal rank.
    pub reciprocal_rank: f64,
    /// NDCG at k.
    pub ndcg_at_k: f64,
}

/// Evaluation response.
#[derive(Debug, Clone, Deserialize)]
pub struct EvalResponse {
    /// Collection name.
    pub collection: String,
    /// Number of cases.
    pub total_cases: u32,
    /// Average recall at k.
    pub recall_at_k_avg: f64,
    /// Mean reciprocal rank.
    pub mrr: f64,
    /// Average NDCG at k.
    pub ndcg_at_k_avg: f64,
    /// Per-case results.
    pub per_case: Vec<EvalPerCase>,
}
